### ISSUE TITLE
GetFeature requests broken with MSSQL Driver

### ISSUE THREAD

**Issue by geographika** [mapserver/mapserver#5172]
I have updated my system to using MS7 (with a dev build from http://www.gisinternals.com).

WMS layers are all working fine but all WFS layers are broken. Any GetFeature request returns the following:

```Query error. msMSSQL2008LayerGetItems: tried to find the geometry column in the results from the database, but couldnt find it.  Is it miss-capitialized? 'GEOM'```

My layer configurations have not changed (and work fine in 6.4), and looking at SQL Profiler the initial query to get the column names runs successfully and returns a GEOM field.
I've looked at changes to https://github.com/mapserver/mapserver/blob/branch-7-0/mapmssql2008.c and can't see what would have changed to cause this error.

**Comment by geographika** [comment]
From discussions with @szekerest 

>This issue appears to be a problem in the ODBC driver. Actually SQLDescribeCol returns only 
>the first character of the column names, that's why the geometry column cannot be identified. This is >probably due to a misinterpretation of the widechar strings as single byte strings somewhere in the >driver.

I'm not sure what is meant by ODBC driver -  the underlying Windows ODBC driver, or the driver based on mapmssql2008.c. 

All was working fine (and still is) in MapServer 6.4, so I'm not sure what has changed in version 7.

**Comment by geographika** [comment]
After getting the code to compile and setting up the debugger on Windows I was able to find the cause of this issue. 

It appears there have been changes to the ODBC types since Windows 2003. 

> Some ODBC functions that were previously defined with SQLINTEGER and 
> SQLUINTEGER parameters have been changed where appropriate to use the new SQLLEN and
> SQLULEN typedefs. These changes are listed in the next section, Function Declaration Changes.

https://msdn.microsoft.com/en-us/library/ms716287(v=vs.85).aspx

When debugging the following error was thrown just after the SQLDescribeCol function. 

`Run-Time Check Failure #2 - Stack around the variable 'columnSize' was corrupted.`

Fixing the first type led to the next error:

<img width="344" alt="debug_error" src="https://cloud.githubusercontent.com/assets/490840/13604045/3087ece8-e541-11e5-992f-fa3d04a28081.png">

After changing all the SQLINTEGER and SQLUINTEGER to SQLLEN and SQLULEN respectively, the GetFeature requests worked again. 

I have created a pull request at https://github.com/mapserver/mapserver/pull/5262
